repo/listing.go

```diff
@@ -169,6 +169,11 @@ func CreateListing(r []byte, isTestnet bool, dstore *Datastore, repoPath string)
 		ld.Slug = slug
 	}
 	retListing, err := NewListingFromProtobuf(ld)
+
+	if retListing.Metadata.Version <= 0 {
+		retListing.Metadata.Version = ListingVersion
+	}
+
```

Comment:
I recommended making `NewListingFromProtobuf` responsible for setting reasonable defaults instead of making the caller function do it.